### DISCUSSION TITLE
Loopback for the changes Zend directory

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#4057] [Questions & Answers] [answered]
I have an old store that has mandatory changes in the files located here **/lib/Zend/Locale/Data/{en.xml, ...}**, mainly for currency positions with different languages.

Meanwhile we removed Magento Zend from /app/.code/core and /lib, and OpenMage now requests using Composer to install ZF1-Future. If I change the files in this directory **/vendor/shardj/zf1-future/library/Zend/Locale/Data** they will be replaced with every update.

Do we have a loopback for changes related to the Zend folder? I tried to copy the modified files in both directories /app/code/core/Zend and /lib/Zend without any luck.

**Comment by fballiano** [comment]
composer patches like shown here: https://github.com/fballiano/mcommerce/blob/main/composer.json

<img width="907" alt="Screenshot 2024-06-30 alle 16 20 01" src="https://github.com/OpenMage/magento-lts/assets/909743/5a64190a-b3ee-4489-a13b-568f3a4f1d92">

you can patch anything coming from composer with this method

**Comment by fballiano** [comment, marked as answer]
actually we have them also in openmage's composer.json

<img width="916" alt="Screenshot 2024-06-30 alle 16 20 59" src="https://github.com/OpenMage/magento-lts/assets/909743/2c80de78-e81c-4aad-8d97-3a3c02a32289">

**Comment by addison74** [comment]
Thank you. I will document myself to make a patch, I suspected that this is the solution. If Composer updates, will the patch be applied afterwards every time?

**Comment by fballiano** [comment]
you may have to update the patches but storing a patch makes a porting much easier because you know exactly what lines you're changing, while having a full file makes it more difficult to remember what you tweaked

**Comment by pquerner** [comment]
Maybe this is also the solution to https://github.com/OpenMage/magento-lts/discussions/3462 - atleast for small code changes.

**Reply by addison74** [reply]
Thank you. There are a lot of solutions searching on the Internet. One very simple is as follows

https://tomasvotruba.com/blog/2020/07/02/how-to-patch-package-in-vendor-yet-allow-its-updates
